File: app/src/main/java/me/magnum/melonds/repositories/RomsRepository.java
package me.magnum.melonds.repositories;

import io.reactivex.Observable;
import me.magnum.melonds.model.Rom;
import me.magnum.melonds.model.RomConfig;
import me.magnum.melonds.model.RomScanningStatus;

import java.util.List;

public interface RomsRepository {
    Observable<List<Rom>> getRoms();
    Observable<RomScanningStatus> getRomScanningStatus();

    void updateRomConfig(Rom rom, RomConfig romConfig);
    void rescanRoms();
}

File: app/src/main/java/me/magnum/melonds/repositories/RomsRepository.kt
package me.magnum.melonds.repositories

import io.reactivex.Observable
import me.magnum.melonds.model.Rom
import me.magnum.melonds.model.RomConfig
import me.magnum.melonds.model.RomScanningStatus
import java.util.*

interface RomsRepository {
    fun getRoms(): Observable<List<Rom>>
    fun getRomScanningStatus(): Observable<RomScanningStatus>

    fun updateRomConfig(rom: Rom, romConfig: RomConfig)
    fun setRomLastPlayed(rom: Rom, lastPlayed: Date)
    fun rescanRoms()
}
